features/FEATURE_BLE/ble/gap/Gap.h

```diff
@@ -492,6 +492,37 @@ class Gap {
         {
         }
 
+        /**
+         * Function invoked when the connections changes the ATT_MTU which controls
+         * the maximum size of an attribute that can be read in a single L2CAP packet
+         * which might be fragmented across multiple packets.
+         *
+         * @param connectionHandle The handle of the connection that changed the size.
+         * @param attMtuSize
+         */
+        virtual void onAttMtuChanged(
+            connection_handle_t connectionHandle,
+            uint16_t attMtuSize
+        )
+        {
+        }
+
+        /**
+         * Function invoked when the connections changes the maximum number of octets
+         * that can be sent or received by the controller in a single packet.
+         *
+         * @param connectionHandle The handle of the connection that changed the size.
+         * @param txSize Number of octets we can send on this connection in a single packet.
+         * @param rxSize Number of octets we can receive on this connection in a single packet.
+         */
+        virtual void onPacketPaylodSizeChanged(
```

Comment:
There's a typo in the name: Paylod -> Payload. 
I wonder if the name shouldn't be closer to names in the spec which is `LE Data Length Change Event`. 
Also, what about change of transmission time of packets ? Should we expose that information or expect users to do the math when PHY change ?

Reply:
frankly I don't see the utility of knowing the packet size in the application since there's nothing that can be done about it but I can add a time event if ever requested